docs/contracts.rst

```diff
@@ -284,8 +284,38 @@ It will generate a function of the following form::
         b = data[arg1][arg2][arg3].b;
     }
 
-Note that the mapping in the struct is omitted because there
-is no good way to provide the key for the mapping.
+    .. note::
+        The mapping in the struct is omitted because there is no good way to provide the key for the mapping.
+
+If you have a `public` state variable of array type, then it only returns
+individual elements to reduce the gas cost of returning an entire array. You
+use parameters to specify which individual element to return, for example
+``data(0)``. If you want to return an entire array in one call, then you need
+to write a helper function, for example:
+
+::
+
+  pragma solidity ^0.4.0;
+  contract arrayExample {
+    // public state variable
+    uint[] public myArray;
+
+    // caller function, generated by compiler
```

Comment:
What does `caller` refer to here? Perhaps just `Getter function generated by the compiler`